### ISSUE TITLE
Object not found error / Error code: Wsl/ERROR_NOT_FOUND - wsl can be run and used only with elevated privileges ("run as admin")

### ISSUE THREAD

**Issue by JVsup** [microsoft/WSL#10767]
### Windows Version

Microsoft Windows [Version 10.0.19045.3570]

### WSL Version

2.0.9.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

Linux version 5.15.133.1-microsoft-standard-WSL2 (root@1c602f52c2e4) (gcc (GCC) 11.2.0, GNU ld (GNU Binutils) 2.37) #1 SMP Thu Oct 5 21:02:42 UTC 2023

### Distro Version

Debian 11.8

### Repro Steps

Today wsl stopped working, I cannot acces it - all i am getting is error that object doesnt exist no matter what i do. Error messages are completely unhelpful.

From standart powershell, i cant do anything, but if I run it with elevated privileges as admin, suddenly i can see the distro, run in, use it, etc.

same goes for windows terminal, launching linux apps, and so on. I need to run them as admin, or nothing works.

I have no clue what caused this deranged behavior, because until today I did not need to run anything wsl related as admin...

### Expected Behavior

I should be able to use wsl with standard user privileges, not only as admin.

### Actual Behavior

**standart powershell (run with normal user permisions)**

```
Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

Try the new cross-platform PowerShell https://aka.ms/pscore6

PS C:\Users\janva> wsl -l -v
Prvek nebyl nalezen.
Error code: Wsl/ERROR_NOT_FOUND
PS C:\Users\janva> wsl --install
Ubuntu je již nainstalovaný.
Spouští se Ubuntu…
Installing, this may take a few minutes...
WslRegisterDistribution failed with error: 0x80070490
Error: 0x80070490 Prvek nebyl nalezen.

Press any key to continue...
Operace byla dokončena úspěšně.
PS C:\Users\janva> wsl
Prvek nebyl nalezen.
Error code: Wsl/ERROR_NOT_FOUND
PS C:\Users\janva> wsl.exe
Prvek nebyl nalezen.
Error code: Wsl/ERROR_NOT_FOUND
PS C:\Users\janva> wsl --version
Verze WSL: 2.0.9.0
Verze jádra: 5.15.133.1-1
Verze WSLg: 1.0.59
Verze MSRDC: 1.2.4677
Verze Direct3D: 1.611.1-81528511
Verze DXCore: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Verze Windows: 10.0.19045.3570
PS C:\Users\janva> wsl --status
Prvek nebyl nalezen.
Error code: Wsl/ERROR_NOT_FOUND
```

--------
--------

**powreshell with elevated privileges**

```
PS C:\> wsl -l -v
  NAME      STATE           VERSION
* Debian    Running         2
PS C:\> wsl --status
Výchozí distribuce: Debian
Výchozí verze: 2
PS C:\> wsl --version
Verze WSL: 2.0.9.0
Verze jádra: 5.15.133.1-1
Verze WSLg: 1.0.59
Verze MSRDC: 1.2.4677
Verze Direct3D: 1.611.1-81528511
Verze DXCore: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Verze Windows: 10.0.19045.3570
```

**Comment by SpeedyLom** [comment]
I'm having this exact issue since yesterday (2023-11-15).

**Comment by MMiller7901** [comment]
Also having similar issues since yesterday (2023-11-15) 
wsl has completely quit responding to commands and just hangs PowerShell

_<3>WSL (82) ERROR: CreateProcessEntryCommon:502: execvpe /bin/bash failed 5_

**Comment by aleluis79** [comment]
Same issue here, after yesterday's Windows 10 Update (2023-11-15).

**Comment by keesj-riscure** [comment]
Similar issue here (and X11 window corruption)

# Admin
```
C:\Users\ME>wsl -l -v
  NAME      STATE           VERSION
* Ubuntu    Running         2
```

# User
```
C:\Users\ME>wsl -l -v
Element not found.
Error code: Wsl/ERROR_NOT_FOUND

```

**Comment by SinisterGaze** [comment]
Have got the same issue several times since 2023-11-15. Always the same error when i try to run any wsl command or open my ubuntu distro: 
```
Element not found.
Error code: Wsl/ERROR_NOT_FOUND
```
I am able to reinstall wsl and ubuntu from scratch and use it for a while but my wsl seems to be corrupted if

1. I try to use Docker Desktop (I am using wsl2 backend)
2. I run "wsl --update" after wsl points out there is a new update to wsl

This is my wsl --version print:

```
WSL-version: 2.0.9.0
Kernelversion: 5.15.133.1-1
WSLg-version: 1.0.59
MSRDC-version: 1.2.4677
Direct3D-version: 1.611.1-81528511
DXCore-version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows-version: 10.0.19045.3693
```

**Comment by MMiller7901** [comment]
This is a bit mind-boggling to me that the same issue hit us all on the same day - I had not installed the latest three updates [KB5032189, KB5032339, or KB4023057] when this started, and installing them has not made a difference.

**Comment by SinisterGaze** [comment]
Update:

I now managed to reproduce and fix my issue by uninstalling the last file in the list below from apps & features. I got that file from running "wsl --update". My wsl doesn't crash anymore as long as I avoid running "wsl --update".

![image](https://github.com/microsoft/WSL/assets/22778557/41b0f923-4840-4b08-85b4-e459e7976a37)

**Comment by MMiller7901** [comment]
This fix worked for me! https://github.com/microsoft/WSL/issues/10764#issuecomment-1815485695

**Comment by mhicks-cfs** [comment]
I am seeing the same error, for me it started after installing the Windows update KB5032189. After rebooting WSL started to behave in the same way as the OP.
Unfortunately @SinisterGaze I do not have the same set of WSL installs as you do, none are 8 KB on my system.

**Comment by Ddog800** [comment]
I just noticed this today as well. Looks like WSL updated yesterday from the MS store (reporting version 2.0.9.0). Also installed Windows 10 KB5032189 yesterday as well.

**Comment by RobertOstermann** [comment]
I am also seeing this error with WSL version 2.0.9.0 and installed Windows 10 KB5032189 yesterday.

**Comment by benhillis** [comment]
Thanks for reporting, I'm not seeing /logs in this thread, could you please collect?

**Comment by JVsup** [comment]
> Thanks for reporting, I'm not seeing /logs in this thread, could you please collect?

Here they are, one log for stuff ran as standard user and another with elevated privileges
[WslLogs-2023-11-18_07-26-40_user.zip](https://github.com/microsoft/WSL/files/13399219/WslLogs-2023-11-18_07-26-40_user.zip)
[WslLogs-2023-11-18_07-28-44_admin.zip](https://github.com/microsoft/WSL/files/13399220/WslLogs-2023-11-18_07-28-44_admin.zip)

**Comment by 10cheon00** [comment]
> Update:
> 
> I now managed to reproduce and fix my issue by uninstalling the last file (8kB) in the list below from apps & features. I got that file from running "wsl --update". My wsl doesn't crash anymore as long as I avoid running "wsl --update".
> 
> ![image](https://user-images.githubusercontent.com/22778557/283832547-41b0f923-4840-4b08-85b4-e459e7976a37.png)

This is works for me. Just uninstall latest app. Great!

**Comment by frocentus** [comment]
> > Update:
> > I now managed to reproduce and fix my issue by uninstalling the last file (8kB) in the list below from apps & features. I got that file from running "wsl --update". My wsl doesn't crash anymore as long as I avoid running "wsl --update".
> > ![image](https://user-images.githubusercontent.com/22778557/283832547-41b0f923-4840-4b08-85b4-e459e7976a37.png)
> 
> This is works for me. Just uninstall latest app. Great!

This worked for me also, altough mine had not 8 KByte but around 160 MByte. But after removing it, WSL2 started to work again.

**Comment by VSurin1** [comment]
This helped me
https://winitpro.ru/index.php/2020/07/13/zapusk-linux-v-windows-wsl-2/

**Comment by ghusta** [comment]
I have the same problem with WSL 2.

Maybe it's related with this problem I have when I try to start Docker Desktop :

![image](https://github.com/microsoft/WSL/assets/456285/e1cc924c-57fd-4c8b-a8ef-01d917c07136)

**Comment by eugenylitvinov** [comment]
The same problem for me, work stopped, docker doesn't work, wsl doesn't work. After removing MS Updale for wsl -  wsl starts working but Docker reinstalls this update on installation and everything crashes immediately.

**Comment by Alk3m1st** [comment]
Just had this problem today and found this thread. As others reported, uninstalling the update has fixed the issue for me also.

Hopefully a permanent resolution will be sorted soon to allow the updates.

**Comment by baiciluigi** [comment]
Same issue, can only run wsl with admin rights, everything else fails. Have to manually run it on admin.
Docker Desktop fails, but when launched with admin rights, succeeds aswell.
For now I can work like this but it needs to be fixed..

**Comment by mslinn** [comment]
My problem began after updating the Microsoft Store apps. Windows update shows this:
![image](https://github.com/microsoft/WSL/assets/485818/c76e58f7-df01-4bf1-a2c8-b0102e53db7b)

The store shows this:
![image](https://github.com/microsoft/WSL/assets/485818/85867d13-0090-4916-a1c8-4d58697ef6c0)

Windows Update shows this:
![image](https://github.com/microsoft/WSL/assets/485818/8d6efd89-e9d0-4049-9b3d-e3eb334f8ffd)

How would I roll back the updated WSL installation?

**Comment by mikeputnam** [comment]
Running a Terminal as Administrator seems to work.   This is problematic for the terminal in VSCode however.

**Comment by mslinn** [comment]
@mikeputnam Right, VSCode does not update extensions when running as Administrator. This is not a useful solution.

**Comment by danielhuang-030** [comment]
> The same problem for me, work stopped, docker doesn't work, wsl doesn't work. After removing MS Updale for wsl - wsl starts working but Docker reinstalls this update on installation and everything crashes immediately.

I had the same problem, but I solved it by reinstalling the previous version (4.24.2.0) of docker desktop.

**Comment by 573** [comment]
> > Update:
> > I now managed to reproduce and fix my issue by uninstalling the last file (8kB) in the list below from apps & features. I got that file from running "wsl --update". My wsl doesn't crash anymore as long as I avoid running "wsl --update".
> > ![image](https://user-images.githubusercontent.com/22778557/283832547-41b0f923-4840-4b08-85b4-e459e7976a37.png)
> 
> This is works for me. Just uninstall latest app. Great!

**tl;dr** Currently I only left the entry with `459 MB` still installed.

Same here, after removing entry in Apps & Features as in picture above the one with tinier tux icon (localised label in the picture) `wsl --list -v` was running as normal again.

**Comment by Alkarex** [comment]
@SinisterGaze 's solution worked for me as well on Windows 10 https://github.com/microsoft/WSL/issues/10767#issuecomment-1816540839

**Comment by fabio-r-goncalves** [comment]
So, it seems that @SinisterGaze's solution works. The problem is that now I cannot use docker desktop. It tries to run wsl --update and fails to do so.

**Comment by fabio-r-goncalves** [comment]
Ok, so I think I can have everything working. So, follow @SinisterGaze's instructions and remove the update. Then, if needed, install wsl kernel update from https://www.catalog.update.microsoft.com/Search.aspx?q=wsl

**Comment by FlorianSchaetz** [comment]
Doesn't work for me, since I don't have that 8kb file. Also doing the Kernel Update doesn't help. Tried de-installing everything and re-installing, nothing work. Running Terminal in admin works, but that's not really a solution. Docker Desktop still not functional.

**Comment by 573** [comment]
Just in case it helps someone:
The steps I took where re-installing the Windows-Store version (so basically getting all three icons above in the famous screen above) after I took this exact measure:

> Thank you @pahar0. This confirms our theory: There is an old WSL installation that's conflicting with the latest package.
> 
> Can you try to:
> 
>     * Delete: `HKEY_LOCAL_MACHINE\SOFTWARE\Classes\PackagedCom\ClassIndex\{A9B7A1B9-0671-405C-95F1-E0612CB4CE7E}`
> 
>     * See if that solves the issue
> 
> 
> (You can use the .reg you just captured as backup if needed)

_Originally posted by @OneBlue in https://github.com/microsoft/WSL/issues/10782#issuecomment-1820066209_

**Comment by aleluis79** [comment]
![imagen](https://github.com/microsoft/WSL/assets/3486008/917358b2-4e99-4448-874d-3dc737d5734f)

I added a dash to the old keys. That worked for me.

**Comment by Ruansh233** [comment]
> Update:
> 
> I now managed to reproduce and fix my issue by uninstalling the last file (8kB) in the list below from apps & features. I got that file from running "wsl --update". My wsl doesn't crash anymore as long as I avoid running "wsl --update".
> 
> ![image](https://user-images.githubusercontent.com/22778557/283832547-41b0f923-4840-4b08-85b4-e459e7976a37.png)

I have the same 8kB file, but it doesn't work after I delete it. So sad.

**Comment by 573** [comment]
> > Update:
> > I now managed to reproduce and fix my issue by uninstalling the last file (8kB) in the list below from apps & features. I got that file from running "wsl --update". My wsl doesn't crash anymore as long as I avoid running "wsl --update".
> > ![image](https://user-images.githubusercontent.com/22778557/283832547-41b0f923-4840-4b08-85b4-e459e7976a37.png)
> 
> I have the same 8kB file, but it doesn't work after I delete it. So sad.

Have you tried commenting / deleting the registry keys for old WSL mentioned further up ?

**Comment by Ruansh233** [comment]
> > > Update:
> > > I now managed to reproduce and fix my issue by uninstalling the last file (8kB) in the list below from apps & features. I got that file from running "wsl --update". My wsl doesn't crash anymore as long as I avoid running "wsl --update".
> > > ![image](https://user-images.githubusercontent.com/22778557/283832547-41b0f923-4840-4b08-85b4-e459e7976a37.png)
> > 
> > 
> > I have the same 8kB file, but it doesn't work after I delete it. So sad.
> 
> Have you tried commenting / deleting the registry keys for old WSL mentioned further up ?

The weird thing is I don't have such registry keys. I have tried all solutions under the issue. But none of them works for my case. I will report later if I find sth.

**Comment by craigloewen-msft** [comment]
Hey folks if you're seeing this error can you please try upgrading to [version 2.0.11](https://github.com/microsoft/WSL/releases/tag/2.0.11) which should have a fix! You can do this by running `wsl --update --pre-release` from a command window with admin privileges, and you can check your version with `wsl --version`.

Please let us know if that fixed this issue, thank you!

**Comment by o-sdn-o** [comment]
Thank you! `wsl --update --pre-release` fixed my wsl setup (10.0.19045.3693).

**Comment by Ruansh233** [comment]
> Hey folks if you're seeing this error can you please try upgrading to [version 2.0.11](https://github.com/microsoft/WSL/releases/tag/2.0.11) which should have a fix! You can do this by running `wsl --update --pre-release` from a command window with admin privileges, and you can check your version with `wsl --version`.
> 
> Please let us know if that fixed this issue, thank you!

excellent!!!!

**Comment by dokmaister** [comment]
> Ok, so I think I can have everything working. So, follow @SinisterGaze's instructions and remove the update. Then, if needed, install wsl kernel update from https://www.catalog.update.microsoft.com/Search.aspx?q=wsl

My version of the WSL I had to uninstall was something like 168MB and not 8KB (the one with the "smaller" icon). 
The wsl kernel update was needed on my side as well. Thanks for pointing this out!
So both steps were needed to get it back up and running.

**Comment by MMiller7901** [comment]
Uninstalling the 8kb file listed above gives an error when running wsl of "The specified file is encrypted and the user does not have the ability to decrypt it."
and updating to 2.0.11 spits another new error "wsl: Failed to create network endpoint..."

Any clues on where I should go from here?

![error](https://github.com/microsoft/WSL/assets/151060091/23ab1083-e9d6-4fe2-a5d7-357bb66250d7)

**Comment by keesj-riscure** [comment]
> Hey folks if you're seeing this error can you please try upgrading to [version 2.0.11](https://github.com/microsoft/WSL/releases/tag/2.0.11) which should have a fix! You can do this by running `wsl --update --pre-release` from a command window with admin privileges, and you can check your version with `wsl --version`.
> 
> Please let us know if that fixed this issue, thank you!

I had the run the command 3 times (It was stuck the first time, appeared to have worked the second time) but after a reboot the problem remained present. updating a 3rd time did the trick. Thank you
![image](https://github.com/microsoft/WSL/assets/129783850/f8d1f2dd-63ee-4dc7-aa97-b2c8de32610c)

**Comment by mhicks-cfs** [comment]
> Hey folks if you're seeing this error can you please try upgrading to [version 2.0.11](https://github.com/microsoft/WSL/releases/tag/2.0.11) which should have a fix! You can do this by running `wsl --update --pre-release` from a command window with admin privileges, and you can check your version with `wsl --version`.
> 
> Please let us know if that fixed this issue, thank you!

This worked for me, thanks!
